src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -77,70 +117,99 @@ public void tearDownAfterTest() {
+        ArrayList<ItineraryExistence.ItineraryExistenceResult> validDays = Lists.newArrayList(
+            existence.thursday,
+            existence.saturday,
+            existence.sunday,
+            existence.monday,
+            existence.tuesday
+        );
+        // Check (and remove) the extra invalid day if we inserted one above.
+        if (insertInvalidDay) {
+            Assertions.assertFalse(validDays.get(INVALID_DAY_INDEX).isValid());
+            validDays.remove(INVALID_DAY_INDEX);
+        }
+
         // FIXME: For now, just check that the first itinerary in the list is valid. If we expand our check window from
         //  7 days to 14 (or more) days, this may need to be adjusted.
-        Assertions.assertTrue(existence.monday.isValid());
-        Assertions.assertTrue(ItineraryUtils.itinerariesMatch(expectedItinerary, existence.monday.itineraries.get(0)));
-        Assertions.assertTrue(existence.tuesday.isValid());
-        Assertions.assertTrue(ItineraryUtils.itinerariesMatch(expectedItinerary, existence.tuesday.itineraries.get(0)));
-        Assertions.assertTrue(existence.thursday.isValid());
-        Assertions.assertTrue(
-            ItineraryUtils.itinerariesMatch(expectedItinerary, existence.thursday.itineraries.get(0))
-        );
-        Assertions.assertTrue(existence.saturday.isValid());
-        Assertions.assertTrue(
-            ItineraryUtils.itinerariesMatch(expectedItinerary, existence.saturday.itineraries.get(0))
-        );
-        Assertions.assertTrue(existence.sunday.isValid());
-        Assertions.assertTrue(ItineraryUtils.itinerariesMatch(expectedItinerary, existence.sunday.itineraries.get(0)));
+        for (ItineraryExistence.ItineraryExistenceResult validDay : validDays) {
+            Assertions.assertTrue(validDay.isValid());
+            Assertions.assertTrue(ItineraryUtils.itinerariesMatch(expectedItinerary, validDay.itineraries.get(0)));
+        }
 
+        // When the check is not requested for a day, the existence entry will be null.
         Assertions.assertNull(existence.wednesday);
         Assertions.assertNull(existence.friday);
     }
 
+    private static Stream<Arguments> createCheckAllItinerariesExistTestCases() {
+        return Stream.of(
+            Arguments.of(false, "checkAllDays = false should produce allCheckedDaysAreValid = true."),
+            Arguments.of(true, "checkAllDays = true should produce allCheckedDaysAreValid = false.")
+        );
+    }
+
     /**
-     * Test case in which at least one itinerary does not exist,
-     * and therefore result.allCheckedDatesAreValid should be false.
+     * @return The new {@link Date} object with the date portion set to the specified {@link LocalDate} in OTP timezone.
      */
-    @Test
-    public void canCheckAtLeastOneTripDoesNotExist() throws URISyntaxException {
-        MonitoredTrip trip = makeTestTrip();
+    public static Date getNewItineraryDate(Date itineraryDate, LocalDate date) {
```

Comment:
Same comment as https://github.com/ibi-group/otp-middleware/pull/102#discussion_r534843634.

Reply:
Actually, maybe this is ok to keep in this file since it isn't used elsewhere.